ImputationPipeline/PRSWrapper.wdl

```diff
@@ -0,0 +1,179 @@
+version 1.0
+import "ScoringPart.wdl" as Score
+
+workflow PRSWrapper {
+  input {
+    Array[String] condition_names
+    Array[Boolean] score_condition
+    Array[Float] percentile_thresholds
+    Array[File] weights_files
+
+    File vcf
+    String sample_id
+    Boolean redoPCA = false
+
+    File population_loadings
+    File population_meansd
+    File population_pcs
+    File pruning_sites_for_pca # and the sites used for PCA
+    File population_vcf
+    String population_basename
+  }
+
+  if (length(condition_names) != length(score_condition) || length(condition_names) != length(weights_files) || length(condition_names) != length(percentile_thresholds)) {
+    call ErrorWithMessage {
+      input:
+        message = "conditions_names, use_condition, use_ancestry_correction, and weights_files must all be arrays of the same length"
+    }
+  }
+
+  scatter(i in range(length(condition_names))) {
+    if (score_condition[i]) {
+      call Score.ScoringImputedDataset {
+        input:
+          weights = weights_files[i],
+          imputed_array_vcf = vcf,
+          population_loadings = population_loadings,
+          population_meansd = population_meansd,
+          population_pcs = population_pcs,
+          pruning_sites_for_pca = pruning_sites_for_pca,
+          population_vcf = population_vcf,
+          basename = sample_id,
+          population_basename = population_basename,
+          redoPCA = redoPCA
+      }
+
+      call SelectValuesOfInterest {
+        input:
+          score_result = select_first([ScoringImputedDataset.adjusted_array_scores]),
+          sample_id = sample_id,
+          condition_name = condition_names[i],
+          threshold = percentile_thresholds[i]
+      }
+    }
+
+    if (!score_condition[i])
+    {
+      call CreateUnscoredResult {
+        input:
+          sample_id = sample_id,
+          condition_name = condition_names[i]
+      }
+    }
+
+    File result_for_condition = select_first([SelectValuesOfInterest.results, CreateUnscoredResult.results])
+  }
+
+  call JoinResults{
+    input:
+      results_in = result_for_condition
+  }
+
+  output {
+    File results = JoinResults.results
+  }
+}
+
+task SelectValuesOfInterest {
+  input {
+    File score_result
+    String sample_id
+    String condition_name
+    Float threshold
+  }
+
+  command <<<
+    Rscript - <<- "EOF"
+    library(dplyr)
+    library(readr)
+    score <- read_tsv("~{score_result}")
+    if (nrow(score) != 1) {
+      quit(1)
+    }
+    if ((score %>% pull(`IID`))[[1]] != "~{sample_id}") {
+      quit(1)
+    }
+
+    raw_score <- (score %>% pull(SCORE1_SUM))[[1]]
+    adjusted_score <- (score %>% pull(adjusted_score))[[1]]
+    percentile <- (score %>% pull(percentile))[[1]]
+
+    result <- tibble(sample_id = "~{sample_id}", ~{condition_name}_raw = raw_score, ~{condition_name}_adjusted = adjusted_score, ~{condition_name}_high = (percentile > ~{threshold}))
```

Comment:
dear R developers: i wish you didn't call them tibbles. it kinda sounds like kibble. or dribble. it's weird.

Reply:
lol